tracer/src/Datadog.Trace/IAST/Analyzers/HardcodedSecretsAnalyzer.cs

```diff
@@ -0,0 +1,210 @@
+// <copyright file="HardcodedSecretsAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Serilog.Core;
+
+namespace Datadog.Trace.Iast.Analyzers;
+
+internal class HardcodedSecretsAnalyzer
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<HardcodedSecretsAnalyzer>();
+    private static HardcodedSecretsAnalyzer? _instance = null;
+
+    private static bool _started = false;
+
+    private static List<SecretRegex>? _secretRules = null;
+
+    private static ManualResetEventSlim _waitEvent = new ManualResetEventSlim(false);
+
+    public HardcodedSecretsAnalyzer()
+    {
+        Log.Debug("HardcodedSecretsAnalyzer -> Init");
+        LifetimeManager.Instance.AddShutdownTask(RunShutdown);
+        _started = true;
+        Task.Run(() => PoolingThread());
+    }
+
+    private static void PoolingThread()
+    {
+        try
+        {
+            Log.Debug("HardcodedSecretsAnalyzer polling thread -> Started");
+            while (_started)
+            {
+                var userStrings = new UserStringInterop[100];
```

Comment:
minor: is it worth putting `100` in  a constant, since it controls the size of the batch.

Reply:
Fixed in 67a5816e9aaef3a8493b7393bb29572cbdc38aa9